e2e/app/element/base-element.ts

```diff
@@ -1,77 +1,53 @@
 import {ClickOptions, ElementHandle, Page, WaitForSelectorOptions} from 'puppeteer';
+import Container from 'app/container';
 
 /**
  * BaseElement represents a web element in the DOM.
  * It implements useful methods for querying and interacting with this element.
  */
-export default class BaseElement {
+export default class BaseElement extends Container {
 
-  static asBaseElement(page: Page, elem: ElementHandle): BaseElement {
-    return new BaseElement(page, elem);
+  static asBaseElement(page: Page, elementHandle: ElementHandle): BaseElement {
```

Comment:
Generally methods starting with `as` aren't static. Can this go in the Page class?